roles/matrix-synapse/tasks/ext/appservice-discord/setup.yml

```diff
@@ -0,0 +1,73 @@
+---
+- name: Ensure Appservice Discord image is pulled
+  docker_image:
+        name: "{{ matrix_appservice_discord_docker_image }}"
+  when: "matrix_appservice_discord_enabled"
+
+- name: Ensure Appservice Discord configuration path exists
+  file:
+    path: "{{ matrix_appservice_discord_base_path }}"
+    state: directory
+    mode: 0750
+    owner: "{{ matrix_user_username }}"
+    group: "{{ matrix_user_username }}"
+  when: "matrix_appservice_discord_enabled"
+
+- name: Ensure Matrix Appservice Discord config installed
+  copy:
+    content: "{{ matrix_appservice_discord_configuration|to_nice_yaml }}"
+    dest: "{{ matrix_appservice_discord_base_path }}/config.yaml"
+    mode: 0644
+    owner: "{{ matrix_user_username }}"
+    group: "{{ matrix_user_username }}"
+  when: "matrix_appservice_discord_enabled"
+
+- name: Ensure matrix-appservice-discord.service installed
+  template:
+    src: "{{ role_path }}/templates/ext/appservice-discord/systemd/matrix-appservice-discord.service.j2"
+    dest: "/etc/systemd/system/matrix-appservice-discord.service"
+    mode: 0644
+  when: "matrix_appservice_discord_enabled"
+
```

Comment:
Adding a new task here would be useful:

```yaml
- name: Ensure systemd reloaded after matrix-appservice-discord.service installation
  service:
    daemon_reload: yes
  when: "matrix_appservice_discord_enabled and matrix_appservice_discord_systemd_service_result.changed"
```

This brings it up to date with what we've done yesterday in 45618679f519408b59.
That is, it makes systemd automatically get reloaded when the `.service` file gets updated.